systemd/systemd.go

```diff
@@ -589,6 +593,33 @@ func (s *systemd) getGlobalUserStatus(unitNames ...string) ([]*UnitStatus, error
 	return sts, nil
 }
 
+func (s *systemd) CurrentMemoryUsage(unit string) (quantity.Size, error) {
+	out, err := s.systemctl("show", "--property", "MemoryCurrent", unit)
+	if err != nil {
+		return 0, osutil.OutputErr(out, err)
+	}
+	// strip the MemoryCurrent= from the output
+	splitVal := strings.SplitN(strings.TrimSpace(string(out)), "=", 2)
+	if len(splitVal) != 2 {
+		return 0, fmt.Errorf("invalid format from systemd")
+	}
+
+	trimmedVal := strings.TrimSpace(splitVal[1])
+
+	// if the unit is inactive or doesn't exist, the memory usage can be
+	// reported as "[not set]"
+	if trimmedVal == "[not set]" {
+		return 0, fmt.Errorf("memory usage unavailable")
+	}
+
+	intVal, err := strconv.Atoi(trimmedVal)
+	if err != nil {
+		return 0, err
```

Comment:
Maybe add a test-case for this failure so that we can see that the error string gives enough good information ? i.e. that it also include the incorrect string that could not be converted so that bugreports are useful.

Reply:
I suppose some wrapping would be in order for this error, as I'm not sure the caller should wrap error from here as they are mostly unexpected

Reply:
the error message currently includes the string being parsed but is maybe overly specific? 

```
strconv.Atoi: parsing \"blahhhhhhhhhhhh\": invalid syntax
```

I think maybe something like the below is nicer:
```
invalid property format for memory: cannot parse \"blahhhhhhhhhhhh\" as an integer
```

thoughts?